nncf/experimental/openvino_native/graph/nncf_graph_builder.py

```diff
@@ -65,26 +96,47 @@ def create_nncf_graph(model: ov.Model) -> NNCFGraph:
         :return: NNCFGraph.
         """
         nncf_graph = NNCFGraph()
+        visited = set()
+        inference_nodes = []
 
-        for param in model.get_parameters():
+        for param in sorted(model.get_parameters(), key=lambda param: param.get_friendly_name()):
             nncf_graph.add_nncf_node(node_name=param.get_friendly_name(),
                                      node_type=NNCFGraphNodeType.INPUT_NODE,
                                      node_metatype=InputNoopMetatype)
+            visited.add(param.get_friendly_name())
+            for inp in sorted(param.output(0).get_target_inputs(), key=lambda inp: inp.get_node().get_friendly_name()):
+                inference_nodes.append(inp.get_node())
 
-        for result in model.get_results():
+        for result in sorted(model.get_results(), key=lambda result: result.get_friendly_name()):
             nncf_graph.add_nncf_node(node_name=result.get_friendly_name(),
                                      node_type=NNCFGraphNodeType.OUTPUT_NODE,
                                      node_metatype=OutputNoopMetatype)
+            visited.add(result.get_friendly_name())
+
+        while inference_nodes:
```

Comment:
Do you build the inference graph without constants?

Reply:
Initially yes, but below I add constants.